README.md

```diff
@@ -1,35 +1,48 @@
 
-* `docker build -t ww4ff .`
+1. `git clone https://github.com/castorini/howl && cd howl`
 
-* `./start.sh` will start up the docker. (if docker container is running it will use the same container) Note that /data and current directory will be mounted
+2. Install [PyTorch](https://pytorch.org) by following your platform-specific instructions.
 
-* `conda activate ww4ff` 
+3. `pip install -r requirements.txt`
 
-* set `DATASET_PATH` env variable (`COMMON_VOICE_DATASET_PATH` and `WAKE_WORD_DATASET_PATH` might need to be set if processed dataset is missing)
+### Preparing a Dataset
 
+In the example that follows, we describe how to train a custom detector for the word, "fire."
 
-## Pretraining the model
+1. Download a supported data source. We recommend [Common Voice](https://commonvoice.mozilla.org/) for its breadth and free license.
+2. To provide alignment for the data, install [Montreal Forced Aligner](https://montreal-forced-aligner.readthedocs.io/en/latest/installation.html) (MFA)
+and download an [English pronunciation dictionary](http://svn.code.sf.net/p/cmusphinx/code/trunk/cmudict/cmudict-0.7b).
+3. Create a positive dataset containing the keyword: 
+```bash
+INFERENCE_SEQUENCE=[0] DATASET_PATH=data/fire-positive python -m howl.run.create_raw_dataset --negative-pct 0 --vocab fire -i ~/path/to/common-voice --positive-pct 100
 ```
-source envs/pretrain.env
-NUM_LABELS=10 python -m ww4ff.run.pretrain --model res8 --workspace workspace/<workspace_name>
+4. Create a negative dataset without the keyword:
+```bash
+INFERENCE_SEQUENCE=[0] DATASET_PATH=data/fire-negative python -m howl.run.create_raw_dataset --negative-pct 5 --vocab fire -i ~/path/to/common-voice --positive-pct 0
 ```
-
-## Training res8 for hey firefox
-```
-source envs/res8.env
-INFERENCE_SEQUENCE=[0,1] INFERENCE_WEIGHTS=[1,1,1,1,1,1,1,1,1,1] WW4FF_LOG_LEVEL=INFO NUM_LABELS=10 python -m ww4ff.run.train --model res8 --workspace workspace/<workspace_name> --load-weights --vocab " hey" "fire fox"
-
+5. Generate some mock alignment for the negative set, where we don't care about alignment:
+```bash
+DATASET_PATH=data/fire-negative python -m howl.run.attach_alignment --align-type stub
 ```
-
-## Exporting the trained model as js file
+6. Use MFA to generate alignment for the positive set:
+```bash
+mfa_align data/fire-positive eng.dict pretrained_models/english.zip output-folder
 ```
-python -m ww4ff.run.export_honkling -i workspace/<workspace_name>/model-best.pt -o <output_filename>.js --name RES8
+7. Attach the MFA alignment to the positive dataset:
+```bash
+DATASET_PATH=data/fire-positive python -m howl.run.attach_alignment --align-type mfa -i output-folder
 ```
+
+### Training and Running a Model
+
+1. Source the relevant environment variables for training the `res8` model: `source envs/res8.env`.
+2. Train the model: `python -m howl.run.train -i data/fire-negative data/fire-positive --model res8 --vocab fire --workspace workspaces/fire-res8`.
```

Comment:
do we no longer need pretraining?

Reply:
I didn't include that in the directions for simplicity.